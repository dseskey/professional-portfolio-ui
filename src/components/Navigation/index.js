import React, {useEffect} from 'react';
import { Col, Navbar, Nav } from 'react-bootstrap';
import {capitalizeFirstLetter} from '../../utils/helpers';
function Navigation(props) {
    const {
        setActivePage,
        activePage
      } = props;   
      
      useEffect(() => {
        document.title = 'Dan Seskey - ' + capitalizeFirstLetter(activePage);
      }, [activePage]);

    return (
        <Col data-testid="navigation-bar" id="navigation-bar">
        <Navbar collapseOnSelect expand="lg" >
            <Navbar.Brand id='nav-branding-name'eventkey="about" onClick={() => setActivePage('about')}>Dan Seskey</Navbar.Brand>
            <Navbar.Toggle aria-controls="basic-navbar-nav" />
            <Navbar.Collapse id="basic-navbar-nav">
                <Nav className="mr-auto" onSelect={(selectedKey) => setActivePage(selectedKey)}>
                    <Nav.Link eventKey="about"><span className={activePage==='about' ? 'activeSection' : 'inactiveSection'}>About Me</span></Nav.Link>
                    <Nav.Link eventKey="portfolio"><span data-testid="portfolio-nav-item" className={activePage==='portfolio' ? 'activeSection' : 'inactiveSection'}>Portfolio</span></Nav.Link>
                    <Nav.Link eventKey="contact"><span className={activePage==='contact' ? 'activeSection' : 'inactiveSection'}>Contact</span></Nav.Link>
                    <Nav.Link eventKey="resume"><span className={activePage==='resume' ? 'activeSection' : 'inactiveSection'}>Resume</span></Nav.Link>
                </Nav>
            </Navbar.Collapse>
        </Navbar>
        </Col>
    )
}

export default Navigation;