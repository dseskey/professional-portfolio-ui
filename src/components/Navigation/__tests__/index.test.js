import React from 'react';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Navigation from '..'

afterEach(cleanup);

const activePage = 'about';
const mockSetActivePage = jest.fn();

describe("Header component", () =>{
    it('renders', () => {
        render(<Navigation activePage={activePage} setActivePage={mockSetActivePage}/>);
    });
    it('matches snapshot DOM node structure', () => {
        const {asFragment} = render(<Navigation activePage={activePage} setActivePage={mockSetActivePage}/>);
        expect(asFragment()).toMatchSnapshot();
    });
    
});
