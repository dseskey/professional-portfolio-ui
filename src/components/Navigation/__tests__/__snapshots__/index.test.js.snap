// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="col"
    data-testid="navigation-bar"
    id="navigation-bar"
  >
    <nav
      class="navbar navbar-expand-lg navbar-light"
    >
      <span
        class="navbar-brand"
        eventkey="about"
      >
        Dan Seskey
      </span>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          class="mr-auto navbar-nav"
        >
          <a
            class="nav-link"
            data-rb-event-key="about"
            href="#"
            role="button"
          >
            <span
              class="activeSection"
            >
              About Me
            </span>
          </a>
          <a
            class="nav-link"
            data-rb-event-key="portfolio"
            href="#"
            role="button"
          >
            <span
              class="inactiveSection"
              data-testid="portfolio-nav-item"
            >
              Portfolio
            </span>
          </a>
          <a
            class="nav-link"
            data-rb-event-key="contact"
            href="#"
            role="button"
          >
            <span
              class="inactiveSection"
            >
              Contact
            </span>
          </a>
          <a
            class="nav-link"
            data-rb-event-key="resume"
            href="#"
            role="button"
          >
            <span
              class="inactiveSection"
            >
              Resume
            </span>
          </a>
        </div>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
